src/o365/graph/commands/teams/teams-clone.ts

```diff
@@ -0,0 +1,187 @@
+import * as request from 'request-promise-native';
+import auth from '../../GraphAuth';
+import Utils from '../../../../Utils';
+import config from '../../../../config';
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import { CommandOption, CommandValidate } from '../../../../Command';
+import GraphCommand from '../../GraphCommand';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  teamId: string;
+  displayName: string;
+  mailNickname: string;
+  partsToClone: string;
+  description?: string;
+  classification?: string;
+  visibility?: string;
+}
+
+class GraphTeamsCloneCommand extends GraphCommand {
+  public get name(): string {
+    return `${commands.TEAMS_CLONE}`;
+  }
+
+  public get description(): string {
+    return 'Creates a copy of a Microsoft Teams team';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.description = typeof args.options.description !== 'undefined';
+    telemetryProps.classification = typeof args.options.classification !== 'undefined';
+    telemetryProps.visibility = typeof args.options.visibility !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    auth
+      .ensureAccessToken(auth.service.resource, cmd, this.debug)
+      .then((): request.RequestPromise => {
+
+        const requestOptions: any = {
+          url: `${auth.service.resource}/v1.0/teams/${encodeURIComponent(args.options.teamId)}/clone`,
+          headers: Utils.getRequestHeaders({
+            authorization: `Bearer ${auth.service.accessToken}`,
+            "content-type": "application/zip",
```

Comment:
Any reason to use "application/zip"? [The document for the Teams clone](https://docs.microsoft.com/en-us/graph/api/team-clone?view=graph-rest-1.0#request-headers) suggest that `application/json` to be used with the headers.

Reply:
Sorry, My bad. Fixed this.